src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -723,11 +723,17 @@ private void GetReferenceEquality(BinaryOperatorKind kind, ArrayBuilder<BinaryOp
             BoundExpression left,
             BoundExpression right,
             ArrayBuilder<BinaryOperatorAnalysisResult> results,
-            ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+            ref HashSet<DiagnosticInfo> useSiteDiagnostics,
+            bool skipNativeIntegerOperators = false)
         {
             bool hadApplicableCandidate = false;
             foreach (var op in operators)
             {
+                if (skipNativeIntegerOperators &&
+                    op.Kind.OperandTypes() switch { BinaryOperatorKind.NInt => true, BinaryOperatorKind.NUInt => true, _ => false })
```

Comment:
>false [](start = 124, length = 5)

This could be just my personal preference, but it looks like the use of a switch expression here is making the code less readable and unnecessary verbose by comparison to
```
if (skipNativeIntegerOperators)
{
    switch(op.Kind.OperandTypes())
    {
          case BinaryOperatorKind.NInt:
          case BinaryOperatorKind.NUInt:
                continue;
    }
}
```